sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2525,14 +2525,19 @@ class Dataset[T] private[sql](
 
   /**
    * Returns a new Dataset that contains only the unique rows from this Dataset.
-   * This is an alias for `distinct`.
+   * This is an alias for `distinct` on batch [[Dataset]]. For streaming [[Dataset]], it would show
+   * slightly different behavior. (see below)
    *
    * For a static batch [[Dataset]], it just drops duplicate rows. For a streaming [[Dataset]], it
    * will keep all data across triggers as intermediate state to drop duplicates rows. You can use
    * [[withWatermark]] to limit how late the duplicate data can be and system will accordingly limit
    * the state. In addition, too late data older than watermark will be dropped to avoid any
    * possibility of duplicates.
    *
+   * Note that for a streaming [[Dataset]], this method only returns distinct rows only once,
+   * regardless of the output mode. Spark may convert the `distinct` operation to aggregation`,
```

Comment:
> Spark may convert the `distinct` operation to aggregation`

Can we add an example to the user? This question applies to other places.

Reply:
Probably I need to reword, as the behavior is tied to the catalyst rules, which can be changed in anytime. It would be just enough that `, which the behavior may not be same with using `distinct` in SQL statement`. I'll fix it.

Reply:
+1 for the reword version, actually the original comments also confusing, we need to emphasize the distinct clause in SQL.

Reply:
+1 to emphasize the distinct clause in SQL.